builtin/game/misc.lua

```diff
@@ -74,18 +74,38 @@ function core.after(time, func, ...)
 	}
 end
 
-function core.check_player_privs(name, privs)
+function core.check_player_privs(player_or_name, ...)
+	local name = player_or_name
+	-- Check if we have been provided with a Player object.
+	if name.get_player_name ~= nil then
+		name = name:get_player_name()
+	end
+	
+	local requested_privs = {...}
 	local player_privs = core.get_player_privs(name)
 	local missing_privileges = {}
-	for priv, val in pairs(privs) do
-		if val
-		and not player_privs[priv] then
-			table.insert(missing_privileges, priv)
+	
+	if type(requested_privs[1]) == "table" then
+		-- We were provided with a table like { privA = true, privB = true }.
+		for priv, val in pairs(requested_privs[1]) do
+			if val
+			and not player_privs[priv] then
+				table.insert(missing_privileges, priv)
+			end
+		end
+	else
+		-- Only a list, we can process it directly.
+		for index, priv in ipairs(requested_privs) do
```

Comment:
ipairs orders elements, pairs doesn't order them first, and is therefore faster. Only use ipairs if needed.